File: fibonacci/go/fibonacci.go
package main

import "fmt"

func fibonacci() func() int {
	a, b := 0, 1
	return func() int {
		a = a + b
		a, b = b, a
		return a
	}
}

func main() {
	var n int
	fmt.Print("Enter the number of terms: ")
	fmt.Scanf("%d", &n)

	f := fibonacci()
	var fib int
	fmt.Print("Fibonacci Series: ")
	for i := 0; i < n; i++ {
		fmt.Printf("%d, ", fib)
		fib = f()
	}
}

File: fibonacci/c/fibonacci.c
#include <stdio.h>
#include<conio.h>

    int main()
    {
        clrscr(); //this will clear the screen for fresh output & erase the previous output
        int i, n, t1 = 0, t2 = 1, nextTerm;


        printf("Enter the number of terms: ");
        scanf("%d", &n);


        printf("Fibonacci Series: ");


        for (i = 1; i <= n; ++i)  // loop for printing fibonacci series
        {
            printf("%d, ", t1);
            nextTerm = t1 + t2;
            t1 = t2;
            t2 = nextTerm;
        }
        return 0;
    }
